tensorflow/lite/experimental/micro/examples/micro_speech/main.cc

```diff
@@ -49,7 +49,11 @@ int main(int argc, char* argv[]) {
   // The size of this will depend on the model you're using, and may need to be
   // determined by experimentation.
   const int tensor_arena_size = 10 * 1024;
-  uint8_t tensor_arena[tensor_arena_size];
+  #ifdef ESP32
+    uint8_t* tensor_arena = (uint8_t*) malloc(tensor_arena_size);
```

Comment:
Wondering why ESP32 would prefer malloc over static allocation?

Reply:
Either way, we'd like to avoid device-specific directives in `main.cc`. If we do require special behavior for a given device, we should refactor the affected code into a new file and override it with the same mechanism used for things like command_responder.cc, etc.

Reply:
The standard arduino framework for esp32 starts the main "loopTask" with a stack size of 8k:
https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/main.cpp#L27

I believe trying not to use the heap would make the examples more complicated and probably result in some memory debugging if people try to modify the examples and their code breaks unexpectedly. Especially when using this as an arduino library together with other arduino components. I understand why we're trying to avoid this, though.

Reply:
I'm not really sure what to do here.

The micro_vision example will have to get the same treatment. The size of tensor_arena there is already dependent on the platform. And I'd move the allocation inside the main function as it's done here, but that's probably better discussed another time.

I think i need some specific advice on what to do here. I'll also add handling if malloc() fails.

Reply:
@petewarden Do you have any guidance here?